networklibrary/src/main/java/com/eis/smsnetwork/SMSNetworkManager.java

```diff
@@ -38,41 +54,73 @@
         return netDictionary;
     }
 
+    /**
+     * @return A {@link NetSubscriberList} containing Peers who were invited to join the network and
+     * haven't answered yet.
+     */
+    public NetSubscriberList<SMSPeer> getInvitedPeers() {
+        return invitedPeers;
+    }
+
     /**
      * Starts a setResource request to the net
      *
-     * @param key                 The key identifier for the resource.
-     * @param value               The identified value of the resource.
+     * @param key                 The key identifier for the resource. It cannot have a backslash as
+     *                            its last character.
+     * @param value               The identified value of the resource.It cannot have a backslash as
+     *                            its last character.
      * @param setResourceListener Listener called on resource successfully saved or on fail.
      * @author Marco Cognolato
      */
     @Override
-    public void setResource(String key, String value, SetResourceListener<String, String, FailReason> setResourceListener) {
-
+    public void setResource(String key, String value, SetResourceListener<String, String, SMSFailReason> setResourceListener) {
+        boolean hasSucceeded = false;
+        try {
+            CommandExecutor.execute(new SMSAddResource(key, value, netDictionary));
+            hasSucceeded = true;
+        } catch (Exception e) {
+            Log.e(LOG_KEY, "There's been an error: " + e);
+            setResourceListener.onResourceSetFail(key, value, SMSFailReason.MESSAGE_SEND_ERROR);
+        }
+        if (hasSucceeded) setResourceListener.onResourceSet(key, value);
```

Comment:
What about a `return` inside `catch`? So that you're sure that it has succeeded and you can remove three lines for `hasSucceeded`.

Reply:
🤦‍♂